webapp/packages/core-view/src/Menu/MenuService.ts

```diff
@@ -42,20 +41,42 @@ export class MenuService {
     return null;
   }
 
-  addCreator(creator: IMenuItemsCreator): void {
-    this.creators.push(creator);
+  addCreator(creator: IMenuItemsCreatorOptions): void {
+    this.creators.push({
+      ...creator,
+      menus: new Set(creator.menus),
+      contexts: new Set(creator.contexts),
+    });
   }
 
-  setHandler<T = unknown>(handler: IMenuHandler<T>): void {
+  setHandler<T = unknown>(handler: IMenuHandlerOptions<T>): void {
     if (this.handlers.has(handler.id)) {
       throw new Error(`Menu handler with same id (${handler.id}) already exists`);
     }
-    this.handlers.set(handler.id, handler);
+    this.handlers.set(handler.id, {
+      ...handler,
+      menus: new Set(handler.menus),
+      contexts: new Set(handler.contexts),
+    });
   }
 
-  getHandler(context: IDataContextProvider): IMenuHandler | null {
-    for (const handler of this.handlers.values()) {
-      if (handler.isApplicable(context)) {
+  getHandler(contexts: IDataContextProvider): IMenuHandler | null {
+    const menu = contexts.getOwn(DATA_CONTEXT_MENU);
+
+    handlers: for (const handler of this.handlers.values()) {
```

Comment:
seems like this logic duplicates in handlig actions and bindings. I beleive we can unify it in function with this cycle. but may be it is not a bad idea to leave it as it is for flexibility reasons